src/Aquifer.API/Endpoints/Resources/ResourceReferences/Get/Request.cs

```diff
@@ -0,0 +1,7 @@
+namespace Aquifer.API.Endpoints.Resources.ResourceReferences.Get;
+
+public record Request
+{
+    public int ParentResourceId { get; set; }
+    public string? ResourceIdOrExternalId { get; set; }
```

Comment:
I generally wouldn't expect a single property to be two different things, and not only that, but treated as two different types. I think the result of it with the `TryParse` ends up being confusing as a result too. Also notice it's nullable but the Validator restricts that.

Reply:
Thinking about this more, I think the bigger issue is that the data is bad, and that items got inserted with the external id and there was never a second pass to swap them out with the proper Resource Id. The intention wasn't for the external id to be in there. I think the best approach is to script an update that swaps out those external ids with the proper resource id.